src/autolabel/transforms/webpage_transform.py

```diff
@@ -0,0 +1,115 @@
+from autolabel.schema import TransformType
+from autolabel.transforms import BaseTransform
+from typing import List, Dict, Any, Tuple
+import asyncio
+import logging
+import pandas as pd
+import ssl
+
+logger = logging.getLogger(__name__)
+
+MAX_RETRIES = 3
+MAX_KEEPALIVE_CONNECTIONS = 20
+MAX_CONNECTIONS = 100
+BACKOFF = 2
+HEADERS = {}
+HTML_PARSER = "html.parser"
+
+
+class WebpageTransform(BaseTransform):
+    def __init__(
+        self,
+        url_column: str,
+        output_columns: List[str],
+        timeout: int = 5,
+        headers: Dict[str, str] = HEADERS,
+    ) -> None:
+        super().__init__(output_columns)
+        self.url_column = url_column
+        self.headers = headers
+        self.max_retries = MAX_RETRIES
+        try:
+            from bs4 import BeautifulSoup
+            import httpx
+
+            if not headers.get("User-Agent"):
+                from fake_useragent import UserAgent
+
+                headers["User-Agent"] = UserAgent().random
+
+            self.httpx = httpx
+            self.timeout = httpx.Timeout(timeout)
+            limits = httpx.Limits(
+                max_keepalive_connections=MAX_KEEPALIVE_CONNECTIONS,
+                max_connections=MAX_CONNECTIONS,
+                keepalive_expiry=timeout,
+            )
+            self.client = httpx.AsyncClient(
+                timeout=self.timeout, limits=limits, follow_redirects=True
+            )
+            self.client_with_no_verify = httpx.AsyncClient(
+                timeout=self.timeout, limits=limits, follow_redirects=True, verify=False
+            )
+            self.beautiful_soup = BeautifulSoup
+        except ImportError:
+            raise ImportError(
+                "BeautifulSoup, httpx and fake_useragent are required to use the webpage transform. Please install them with the following command: pip install bs4 httpx fake_useragent"
+            )
+
+    def name(self) -> str:
+        return TransformType.WEBPAGE_TRANSFORM
+
+    def _load_metadata(self, url, soup) -> Dict[str, Any]:
+        metadata = {"url": url}
+        if soup.find("title"):
+            metadata["title"] = soup.find("title").get_text()
+        for meta in soup.find_all("meta"):
+            if meta.get("name") and meta.get("content"):
+                metadata[meta.get("name")] = meta.get("content")
+            elif meta.get("property") and meta.get("content"):
+                metadata[meta.get("property")] = meta.get("content")
+        return metadata
```

Comment:
metadata will be missing some columns in case we dont find them in the page right? How will this be handled by _apply?

Reply:
Right now, we are returning all the meta fields that are present from the webpage. An alternate could be to always look for specific fields, such as the description, encoding, social media tags etc.

Reply:
Makes sense. In case one webpage has a title but the other webpage doesn't have a title we will put nan for the second page under the webpage column. This is being handled by pandas. Tried it out using 

```
import pandas as pd
l = [{'a': 1, 'b': 2}, {'a': 3, 'b': 4}]
pd.DataFrame(l)
   a    b    c
0  1  2.0  NaN
1  3  NaN  4.0
```